### ISSUE TITLE
incorrect unit_of_measurement for home assistant for ZW096 power switch

### ISSUE THREAD

**Issue by brianmay** [zwave-js/zwave-js-ui#3956]
### Checklist

- [X]  I am not using Home Assistant. Or: a developer has told me to come here.
- [X] I have checked the troubleshooting section and my problem is not described there.
- [X] I have read the changelog and my problem is not mentioned there.

### Deploy method

Docker

### Z-Wave JS UI version

9.19.0

### ZwaveJS version

13.3.0

### Describe the bug

Home assistant shows these errors:

```
Oct 14 14:25:59 iot2 homeassistant[192561]: 2024-10-14 14:25:59.902 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.workshop_pump_workshop_pump_electric_power_factor_value (<class 'homeassistant.components.mqtt.sensor.MqttSensor'>) is using native unit of measurement 'Power Factor' which is not a valid unit for the device class ('power') it is using; expected one of ['W', 'kW']; Please update your configuration if your entity is manually configured, otherwise create a bug report at https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue+label%3A%22integration%3A+mqtt%22
Oct 14 14:25:59 iot2 homeassistant[192561]: 2024-10-14 14:25:59.903 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.workshop_pump_workshop_pump_electric_pulse_count_value (<class 'homeassistant.components.mqtt.sensor.MqttSensor'>) is using native unit of measurement 'Pulse count' which is not a valid unit for the device class ('power') it is using; expected one of ['W', 'kW']; Please update your configuration if your entity is manually configured, otherwise create a bug report at https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue+label%3A%22integration%3A+mqtt%22
Oct 14 14:25:59 iot2 homeassistant[192561]: 2024-10-14 14:25:59.904 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.workshop_pump_workshop_pump_electric_kvah_value (<class 'homeassistant.components.mqtt.sensor.MqttSensor'>) is using native unit of measurement 'kVAh' which is not a valid unit for the device class ('power') it is using; expected one of ['W', 'kW']; Please update your configuration if your entity is manually configured, otherwise create a bug report at https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue+label%3A%22integration%3A+mqtt%22
```

I was told this is not a bug in home assistant: https://github.com/home-assistant/core/issues/128325

### To Reproduce

1. Add ZW096 power switch
2. Watch for errors.

(only one of my ZW096 power switch device outputs these values, bit confused here)

### Expected behavior

No errors.

### Additional context

I just noticed my zwave-js-ui is old, don't see anything in changelog about this being fixed, but will try anyway.

**Comment by brianmay** [comment]
Updated to latest zwave-js-ui but nothing changed.

**Comment by robertsLando** [comment]
@brianmay Are you using zwave integration or mqtt discovery?

**Comment by robertsLando** [comment]
Is there a specific reason why you are not using the zwave js integration with zui as server?

**Comment by brianmay** [comment]
I have a lot of my personal code that talks directly using mqtt.

**Comment by robertsLando** [comment]
What should I change in the discovery payload in order to fix the issue so? If a valueId reports power factor or Kvah should I use a different device class? If so which one?

**Comment by brianmay** [comment]
No idea sorry. I asked here: https://github.com/home-assistant/core/issues/128325#issuecomment-2420469228 but the only answer was "It seems the logs are clear on this. Also the sensor docs are clear." which doesn't really help me.

**Comment by brianmay** [comment]
Oh, I see, from: https://www.home-assistant.io/integrations/sensor/

> power_factor: Power factor (unitless), unit may be None or %

Pulse count probably shouldn't have a unit.

kVAh - if kWh not appropriate (not really sure here), then maybe it shouldn't have a unit.

**Comment by brianmay** [comment]
Also I think in all cases the device class == power is wrong.

**Comment by robertsLando** [comment]
Yeah I was reading that too but cannot find the appropriate unit for kVAh, there is `apparent_power` but that only lists `VA`. Could you try to manually edit the discovery payload from UI and re-send it to see what's the correct device class to use foreach unit?

**Comment by robertsLando** [comment]
Also could you paste here a node export please? I see I actually handle this here: 

https://github.com/zwave-js/zwave-js-ui/blob/55318465cf8ab81fb065fbfc11cb44e173067480/api/lib/Constants.ts#L116-L185

But maybe that doesn't cover all cases

**Comment by brianmay** [comment]
Is there somewhere I can send you my node export in private? Rather not send a complete list of my devices to a public forum...

Getting puzzled here, because for the relevant device all I see is:

```
 "16": {
    "name": "Pump",
    "loc": "Workshop"
  },
```

I expected to see more details (???).

**Comment by brianmay** [comment]
I suspect what we would need is:

power factor:  "device_class": "power_factor", delete unit_of_measurement
Pulse count:  delete device_class, delete unit_of_measurement
kVAh:  delete device_class, delete unit_of_measurement

For the last two, there doesn't seem to be any good pre-existing device class. Surprised I can't find a "apparent_energy" class actually. "apparent_power" is measuring power, not energy.

**Comment by brianmay** [comment]
Also I would really like to know why, out of 5 of this devices running the same firmware, only one of them seems to be producing these values. That seems very odd.

**Comment by robertsLando** [comment]
> Is there somewhere I can send you my node export in private? Rather not send a complete list of my devices to a public forum...

PM here: daniel.sorridi@gmail.com

> Getting puzzled here, because for the relevant device all I see is:

Are you clicking on advanced and then debug export?

> I suspect what we would need is:

I need to understand how to detect this properly. @kpine do you have any clue how home assistant handles this?

> Also I would really like to know why, out of 5 of this devices running the same firmware, only one of them seems to be producing these values. That seems very odd.

Rediscover node has no effect on this as this is ZUI values and rediscover node is an action sent to MQTT discovery. You could try to run a new interview of the node but I'm not sure if that would help. Do you know if that device has fw updates available? About this I cc @AlCalzone

**Comment by brianmay** [comment]
> PM here: [daniel.sorridi@gmail.com](mailto:daniel.sorridi@gmail.com)

Done

> Are you clicking on advanced and then debug export?

advanced -> Backup -> Export

> Do you know if that device has fw updates available?

I am not familiar with any fw updates. zwave js ui reports all devices on "FW: v1.4".

I am wondering if the difference is the version of zwave I had installed when I did the interviewing. Either that, or I actually have a different generations what should be the exact same product.

The oldest device seems to have the most metrics (15), the most recent device has the fewest (only 4). Huh?

Wonder if I should re-interview the old device and see if I still get all the metrics back again. But no going back again if I do lose them... Which might kill any chances of testing a fix for the incorrect units.

**Comment by AlCalzone** [comment]
> power factor: "device_class": "power_factor", delete unit_of_measurement
> Pulse count: delete device_class, delete unit_of_measurement
> kVAh: delete device_class, delete unit_of_measurement

Agree for the first two, which are unitless, but this part actually looks like a bug in `node-zwave-js`.
The latter seems to be simply missing in HA, so deleting the unit is more of a workaround than a proper fix.

**Comment by AlCalzone** [comment]
There's an ongoing discussion about the missing units here: https://github.com/home-assistant/architecture/discussions/724

**Comment by brianmay** [comment]
> I think he's looking for the debug exports either under the node's Advanced menu, or from the "Debug Info" panel.

I sent him both of those files to his email also.

**Comment by robertsLando** [comment]
@brianmay Thanks, received!

**Comment by robertsLando** [comment]
@AlCalzone There have been any update on HA discussion about this?

**Comment by AlCalzone** [comment]
Not that I'm aware.
